File: mocks/oppgave-mock/src/main/java/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/AtMostOneOfValidator.java
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering;


import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

import static no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering.CountFieldsMatching.count;

@SuppressWarnings("WeakerAccess")
public class AtMostOneOfValidator implements ConstraintValidator<AtMostOneOf, Object> {
    private String[] fields;

    @Override
    public void initialize(AtMostOneOf atMostOneOf) {
        this.fields = atMostOneOf.fields();
    }

    @Override
    public boolean isValid(Object o, ConstraintValidatorContext constraintValidatorContext) {
        return count(o, fields) <= 1L;
    }
}

File: mocks/oppgave-mock/src/main/kotlin/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/AtMostOneOfValidator.kt
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class AtMostOneOfValidator : ConstraintValidator<AtMostOneOf, Any?> {
    private lateinit var fields: Array<String>

    override fun initialize(atMostOneOf: AtMostOneOf) {
        fields = atMostOneOf.fields
    }

    override fun isValid(o: Any?, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        return countFieldsMatching(o!!, fields) <= 1L
    }
}
